aoc2022/src/main/scala/day16.scala

```diff
@@ -0,0 +1,93 @@
+import scala.io.*
+import aoc2022.Algorithms.Dijkstra.*
+
+import scala.collection.immutable.Queue
+
+/**
+ * 
+ * PART 01:
+ * 
+ * I used Dijkstra to get a graph of all the locations of valves that matter (i.e. have a flow rate > 0). This allowed
+ * the exploreValves() function to quickly walk through the valves. This function walks all the possible routes and 
+ * returns all the end Nodes that track the cumulative score. The function halts if all the valves are opened or if 
+ * the remaining time is 0. Also, Nodes that have a lower score than already existing nodes and fewer time left are 
+ * trimmed.
+ * 
+ * PART 02:
+ * 
+ * Simply used the algorithm of part01 twice, leaving out the opened valves from the first iteration. 
+ * 
+ */
+
+
+object day16 extends App:
+
+  private val day: String =
+    this.getClass.getName.drop(3).init
+
+  private val start1: Long =
+    System.currentTimeMillis
+
+  private val (routes, valves): (Map[String, List[String]], Map[String, Int]) =
+  
+    def parse(s: String): ((String, Int), (String, List[String])) = s match
```

Comment:
Perhaps you can create a case class for this result; the named arguments will lead to less confusion about the meaning of it. I've went a different route by assigning the valves to either me or the elephant. In order to check all assignments I needed to reduce the search space.